### ISSUE TITLE
Does not install on AWS

### ISSUE THREAD

**Issue by antonyRoberts** [hashicorp/consul-helm#17]
Attempt to setup consul via helm using this repo and the directions provided via the blog.
The pods fail with the following.

The pods 
`==> Starting Consul agent...
==> Consul agent running!
           Version: 'v1.2.3'
           Node ID: 'ff2962b2-4429-b113-1ae5-c40f78fb7fb6'
         Node name: 'giggly-mouse-consul-4p9zd'
        Datacenter: 'dc1' (Segment: '')
            Server: false (Bootstrap: false)
       Client Addr: [0.0.0.0] (HTTP: 8500, HTTPS: -1, DNS: 8600)
      Cluster Addr: 100.110.0.1 (LAN: 8301, WAN: 8302)
           Encrypt: Gossip: false, TLS-Outgoing: false, TLS-Incoming: false

==> Log data will now stream in as it occurs:

    2018/10/03 17:50:05 [INFO] serf: EventMemberJoin: giggly-mouse-consul-4p9zd 100.110.0.1
    2018/10/03 17:50:05 [WARN] agent/proxy: running as root, will not start managed proxies
    2018/10/03 17:50:05 [INFO] agent: Started DNS server 0.0.0.0:8600 (udp)
    2018/10/03 17:50:05 [INFO] agent: Started DNS server 0.0.0.0:8600 (tcp)
    2018/10/03 17:50:05 [INFO] agent: Started HTTP server on [::]:8500 (tcp)
    2018/10/03 17:50:05 [INFO] agent: started state syncer
    2018/10/03 17:50:05 [INFO] agent: Retry join LAN is supported for: aliyun aws azure digitalocean gce k8s os packet scaleway softlayer triton vsphere
    2018/10/03 17:50:05 [INFO] agent: Joining LAN cluster...
    2018/10/03 17:50:05 [INFO] agent: (LAN) joining: [giggly-mouse-consul-server-0.giggly-mouse-consul-server.default.svc giggly-mouse-consul-server-1.giggly-mouse-consul-server.default.svc giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc]
    2018/10/03 17:50:05 [WARN] manager: No servers available
    2018/10/03 17:50:05 [ERR] agent: failed to sync remote state: No known Consul servers
    2018/10/03 17:50:05 [WARN] memberlist: Failed to resolve giggly-mouse-consul-server-0.giggly-mouse-consul-server.default.svc: lookup giggly-mouse-consul-server-0.giggly-mouse-consul-server.default.svc on 100.64.0.10:53: no such host
    2018/10/03 17:50:05 [WARN] memberlist: Failed to resolve giggly-mouse-consul-server-1.giggly-mouse-consul-server.default.svc: lookup giggly-mouse-consul-server-1.giggly-mouse-consul-server.default.svc on 100.64.0.10:53: no such host
    2018/10/03 17:50:05 [WARN] memberlist: Failed to resolve giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc: lookup giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc on 100.64.0.10:53: no such host
    2018/10/03 17:50:05 [INFO] agent: (LAN) joined: 0 Err: 3 error(s) occurred:

* Failed to resolve giggly-mouse-consul-server-0.giggly-mouse-consul-server.default.svc: lookup giggly-mouse-consul-server-0.giggly-mouse-consul-server.default.svc on 100.64.0.10:53: no such host
* Failed to resolve giggly-mouse-consul-server-1.giggly-mouse-consul-server.default.svc: lookup giggly-mouse-consul-server-1.giggly-mouse-consul-server.default.svc on 100.64.0.10:53: no such host
* Failed to resolve giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc: lookup giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc on 100.64.0.10:53: no such host
    2018/10/03 17:50:05 [WARN] agent: Join LAN failed: <nil>, retrying in 30s
    2018/10/03 17:50:08 [WARN] manager: No servers available
    2018/10/03 17:50:08 [ERR] http: Request GET /v1/status/leader, error: No known Consul servers from=127.0.0.1:58512
    2018/10/03 17:50:18 [WARN] manager: No servers available
    2018/10/03 17:50:18 [ERR] http: Request GET /v1/status/leader, error: No known Consul servers from=127.0.0.1:58556
    2018/10/03 17:50:27 [WARN] manager: No servers available
    2018/10/03 17:50:27 [ERR] agent: failed to sync remote state: No known Consul servers
    2018/10/03 17:50:28 [WARN] manager: No servers available
    2018/10/03 17:50:28 [ERR] http: Request GET /v1/status/leader, error: No known Consul servers from=127.0.0.1:58614
    2018/10/03 17:50:35 [INFO] agent: (LAN) joining: [giggly-mouse-consul-server-0.giggly-mouse-consul-server.default.svc giggly-mouse-consul-server-1.giggly-mouse-consul-server.default.svc giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc]
    2018/10/03 17:50:36 [INFO] serf: EventMemberJoin: giggly-mouse-consul-server-0 100.126.0.9
    2018/10/03 17:50:36 [INFO] consul: adding server giggly-mouse-consul-server-0 (Addr: tcp/100.126.0.9:8300) (DC: dc1)
    2018/10/03 17:50:36 [INFO] serf: EventMemberJoin: giggly-mouse-consul-server-1 100.104.0.10
    2018/10/03 17:50:36 [INFO] consul: adding server giggly-mouse-consul-server-1 (Addr: tcp/100.104.0.10:8300) (DC: dc1)
    2018/10/03 17:50:36 [WARN] memberlist: Failed to resolve giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc: lookup giggly-mouse-consul-server-2.giggly-mouse-consul-server.default.svc on 100.64.0.10:53: no such host
    2018/10/03 17:50:36 [INFO] agent: (LAN) joined: 2 Err: <nil>
    2018/10/03 17:50:36 [INFO] agent: Join LAN completed. Synced with 2 initial agents
    2018/10/03 17:50:36 [INFO] serf: EventMemberJoin: giggly-mouse-consul-8d92f 100.126.0.8
    2018/10/03 17:50:36 [INFO] serf: EventMemberJoin: giggly-mouse-consul-9bz9r 100.104.0.9
    2018/10/03 17:50:45 [ERR] consul: "Catalog.NodeServices" RPC failed to server 100.126.0.9:8300: rpc error making call: No cluster leader
    2018/10/03 17:50:45 [ERR] agent: failed to sync remote state: rpc error making call: No cluster leader
    2018/10/03 17:50:53 [ERR] consul: "Catalog.NodeServices" RPC failed to server 100.104.0.10:8300: rpc error making call: No cluster leader`

I was and am able to use the repo at https://github.com/helm/charts/tree/master/stable/consul to run traefik with consul.  

Also it is a bit confusing that there exists a chart on helm and one at hashicorp.

**Comment by mitchellh** [comment]
> Also it is a bit confusing that there exists a chart on helm and one at hashicorp.

Yep, we addressed this in our announcement blog but it isn't documented elsewhere so understandable. We plan on migrating to the official chart in time, but during the first releases we wanted to stabilize here.

As for your actual issue: do you have at least 3 distinct hosts for the servers? The most common case I've seen this pop up is that there aren't enough hosts to place the servers (such as minikube, though I know you stated you're in AWS).

We tested the helm chart on Google, Azure, and AWS hosted K8S offerings and verified that it works with the default settings, so its likely something experimental. We also just spun up another EKS cluster and verified it works out of the box.

If you find out more let me know. Unfortunately there isn't anything else to go on at this point...

**Comment by antonyRoberts** [comment]
I have 3 machines in my clusters and the setup is working with the chart hosted on helm's site. I'm not using EKS for kubernetes on AWS

**Comment by mitchellh** [comment]
It's possible that for some reason the Consul servers aren't schedulable on all 3 machines (the default replicas is 3 and that'd require 3 machines). Disabling the affinity rules seems to imply that, because the error messages above mean that certain set members didn't appear.
